backend/inspirehep/records/utils.py

```diff
@@ -96,3 +101,35 @@ def hash_data(data):
     if data:
         return hashlib.md5(data).hexdigest()
     raise ValueError("Data for hashing cannot be empty")
+
+
+def get_pids_for_pid(pid_type, pid_value, provider=None):
+    """Returns all pids registered in PIDStore for record with provided
+    pit_type and pid_value
+
+        If filter_provider id provided then filters results and returns only
+        pids for specified provider
+    Args:
+        pid_type(str): provided pid_type
+        pid_value(str): provided pid_value
+        provider(str): provider for which pids should be returned
+
+    Returns: list of tuples:('pid_type', 'pid_value')
+    """
+    ext_pid = aliased(PersistentIdentifier)
+    pid = aliased(PersistentIdentifier)
+    query = db.session.query(pid.pid_type, pid.pid_value).filter(
+        pid.object_uuid == ext_pid.object_uuid,
+        pid.object_type == ext_pid.object_type,
+        ext_pid.object_type == "rec",
+        ext_pid.pid_type == pid_type,
+        ext_pid.pid_value == pid_value,
+    )
+    if provider:
+        query = query.filter(pid.pid_provider == provider)
+    return query.all()
```

Comment:
why not ``scalar``? it should be one always or ``None``

Reply:
Sorry, but no. You can get more than one DOIs, also when you request for all pids for a record it will be more than one.

Reply:
It was simplified after discussion IRL